### PR TITLE
feature/refactoring statistic

### PR DESCRIPTION
**Pull request by xidealo**

### CODE REVIEW

**Review by MaxAstin — CHANGES_REQUESTED** [review]

File: data/src/main/java/com/bunbeauty/data/repository/AdditionGroupRepository.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        networkConnector.patchAdditionGroup(
+            additionGroupUuid = additionGroupUuid,
+            additionGroupPatchServer = updateAdditionGroup.mapUpdateAdditionGroupServerToPatchAdditionGroup(),
+            token = token
+        )
+
+    }
+
+
+    override suspend fun clearCache() {
```
[the remark this answers is not in the document]
Comment (xidealo) [reply data/src/main/java/com/bunbeauty/data/repository/AdditionGroupRepository.kt:49]: yep

File: data/src/main/java/com/bunbeauty/data/repository/AdditionRepository.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    ) {
+        networkConnector.patchAddition(
+            additionUuid = additionUuid,
+            additionPatchServer = updateAddition.mapUpdateAdditionServerToPatchAddition(),
+            token = token
+        )
+    }
+
+    override suspend fun clearCache() {
+        TODO("Not yet implemented")
```
[the remark this answers is not in the document]
Comment (xidealo) [reply data/src/main/java/com/bunbeauty/data/repository/AdditionRepository.kt:45]: in Vanya's feature

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/additiongrouplist/AdditionGroupListFragment.kt [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+import com.bunbeauty.presentation.feature.additiongrouplist.AdditionGroupListViewModel
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.toPersistentList
+
+@AndroidEntryPoint
+class AdditionGroupListFragment :
+    BaseComposeFragment<AdditionGroupList.ViewDataState, AdditionGroupListViewState, AdditionGroupList.Action, AdditionGroupList.Event>() {
+
+    companion object {
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/additiongrouplist/AdditionGroupListFragment.kt:40]: No need for `companion object`

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/additiongrouplist/AdditionGroupListFragment.kt [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+                    item(
+                        key = R.string.title_position_hidden
+                    ) {
+                        Text(
+                            text = stringResource(id = R.string.title_position_hidden),
+                            style = AdminTheme.typography.titleMedium.bold,
+                            modifier = Modifier.padding(top = 8.dp)
+                        )
+                    }
+                    items(state.hiddenAdditionItems) { hiddenAddition ->
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/additiongrouplist/AdditionGroupListFragment.kt]: keys

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/statistic/ui/StatisticFragment.kt [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+        LazyColumn(
+            contentPadding = PaddingValues(
+                top = 8.dp,
+                start = 16.dp,
+                end = 16.dp,
+                bottom = AdminTheme.dimensions.scrollScreenBottomSpace
+            ),
+            verticalArrangement = Arrangement.spacedBy(8.dp)
+        ) {
+            items(items = statisticViewState.statisticList) { statisticItemModel ->
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/statistic/ui/StatisticFragment.kt:215]: keys

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/additiongrouplist/AdditionGroupListFragment.kt [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+            }.toPersistentList(),
+            isRefreshing = state.isRefreshing,
+            isLoading = state.isLoading
+        )
+    }
+
+    override fun handleEvent(event: AdditionGroupList.Event) {
+        when (event) {
+            AdditionGroupList.Event.Back -> findNavController().popBackStack()
+            is AdditionGroupList.Event.OnAdditionClick -> {
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/additiongrouplist/AdditionGroupListFragment.kt:189]: OnAdditionGroupClick

File: app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/statistic/ui/StatisticFragment.kt [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+                        text = statisticItemModel.proceeds,
+                        style = AdminTheme.typography.bodyMedium.bold,
+                        color = AdminTheme.colors.main.onSurface
+                    )
+                }
+            }
+        }
+    }
+
+    @Preview
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/fooddeliveryadmin/screen/statistic/ui/StatisticFragment.kt:268]: showSystemUi = true

File: data/src/main/java/com/bunbeauty/data/repository/AdditionGroupRepository.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaxAstin) [reply data/src/main/java/com/bunbeauty/data/repository/AdditionGroupRepository.kt:47]: no

**Review by MaxAstin — APPROVED** [review]

**Pull request merged**